internal/common/db.go

```diff
@@ -288,3 +290,39 @@ func ToSqlList(strs []string) string {
 	res = fmt.Sprintf("('%s')", res)
 	return res
 }
+
+func CreateTransactionWithDefer(db *gorm.DB, txSuccess *bool) (func(), *gorm.DB, error) {
```

Comment:
Why is this created here? I don't see it being used in this PR.

Reply:
we have a lot of repeating code for transaction initialization. This function unifies this code. I push it here, so that we can replace all those occurrences in a later PR

Reply:
removed